### ISSUE TITLE
Font error when running scripts if TOilet is installed and not figlet

### ISSUE THREAD

**Issue by bigrichwood** [community-scripts/ProxmoxVE#1092]
### ✅ Have you read and understood the above guidelines?

Yes

### 📜 What is the name of the script you are using?

Proxmox Datacenter Manager LXC

### 📂 What was the exact command used to execute the script?

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/proxmox-datacenter-manager.sh)"

### 📝 Provide a clear and concise description of the issue.

After selecting Yes as the first dialog box to create the LXC, the script fails with a missing font error for Slant. 

In my environment at least this is because I have TOilet installed as a dependecy of Neofetch, this provides a binary for figlet which means the script believes figlet is installed, but it's not compatible with the font Slant. 

### ⚙️ What settings are you using?

- [X] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 🔄 Steps to reproduce the issue.

Run the script, select Yes at the first dialog box. 

### ❌ Paste the full error output (if available).

error: could not load font slant

[ERROR] in line 227: exit code 0: while executing command ascii_art=$(figlet -f slant -w "$term_width" "$APP")

### 🖼️ Additional context (optional).

Kernel Version - Linux 6.8.12-5-pve (2024-12-03T10:26Z)
Manager Version - pve-manager/8.3.1/fb48e850ef9dde2

with TOilet installed:

rich@organa:~$ file /usr/bin/figlet
/usr/bin/figlet: symbolic link to /etc/alternatives/figlet

rich@organa:~$ file /etc/alternatives/figlet
/etc/alternatives/figlet: symbolic link to /usr/bin/figlet-toilet

rich@organa:~$ figlet -v
TOIlet Copyright 2006 Sam Hocevar
Internet: [sam@hocevar.net](mailto:sam@hocevar.net) Version: 0.3, date:

apt install figlet fixes the problem 

rich@organa:~$ file /usr/bin/figlet
/usr/bin/figlet: symbolic link to /etc/alternatives/figlet

rich@organa:~$ file /etc/alternatives/figlet
/etc/alternatives/figlet: symbolic link to /usr/bin/figlet-figlet

rich@organa:~$ figlet -v
FIGlet Copyright (C) 1991-2012 Glenn Chappell, Ian Chai, John Cowan, Christiaan Keet and Claudio Matsuoka
Internet: [info@figlet.org](mailto:info@figlet.org) Version: 2.2.5, date: 31 May 2012

making the test in build.func "command -v figlet-figlet"  may be a way forward, but I'm not sure if that'll work in every environment.

**Comment by oOStroudyOo** [comment]
I was unable to replicate the issue, even with uninstalling figlet and allowing the helper-script to reinstall it:
![image](https://github.com/user-attachments/assets/a2b3d37f-2af6-476f-9973-68bd551eade4)

I am running a slightly newer version of pve-manager/8.3.2/3e76eec21c4a14a7 but there isn't anything in the Proxmox change log  relating to Figlet.

**Comment by bigrichwood** [comment]
Having done a bit more digging, my Proxmox nodes have an old version of figlet which wasn't installed from apt, almost certainly it was installed manually by an earlier helper-script. This version does not support the font slant.

rich@organa:~$ figlet -v
TOIlet Copyright 2006 Sam Hocevar
Internet: <sam@hocevar.net> Version: 0.3, date:

If I install with an apt install figlet, you get a much newer version which does support the appropriate fonts.

rich@organa:~$ figlet -v
FIGlet Copyright (C) 1991-2012 Glenn Chappell, Ian Chai, John Cowan, Christiaan Keet and Claudio Matsuoka
Internet: <info@figlet.org> Version: 2.2.5, date: 31 May 2012

So it looks like simply checking for the existance of figlet in the script isn't enough, it needs to check if it's a new enough version.

**Comment by MickLesk** [comment]
Can you Update your issue description & title with your latest Message? Then i can check it in a few days

**Comment by onethree7** [comment]
> Having done a bit more digging, my Proxmox nodes have an old version of TOilet which includes a figlet binary. This version does not support the font slant, and I think was installed as a dependecy of neofetch.
> 
> rich@organa:~$ figlet -v TOIlet Copyright 2006 Sam Hocevar Internet: [sam@hocevar.net](mailto:sam@hocevar.net) Version: 0.3, date:
> 
> If I install with an apt install figlet, you get a version which does support the appropriate fonts.
> 
> rich@organa:~$ figlet -v FIGlet Copyright (C) 1991-2012 Glenn Chappell, Ian Chai, John Cowan, Christiaan Keet and Claudio Matsuoka Internet: [info@figlet.org](mailto:info@figlet.org) Version: 2.2.5, date: 31 May 2012
> 
> So it looks like simply checking for the existance of figlet in the script isn't enough, it needs to check if it's a suitable version.

you are 100% correct and its reproducable as such. Not creating another issue for #jellyfin . This is everywhere, neofetch is basic to all my LXC.

**Comment by vsisl** [comment]
I seemingly have the same problem when installing [Adguard LXC](https://community-scripts.github.io/ProxmoxVE/scripts?id=adguard).

```bash
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/adguard.sh)"

error: could not load font slant

[ERROR] in line 227: exit code 0: while executing command ascii_art=$(figlet -f slant -w "$term_width" "$APP")
```

**Comment by MickLesk** [comment]
I wonder why you have Toilet or figlet in such ancient versions on the systems? What is the reason?

**Comment by bigrichwood** [comment]
> I wonder why you have Toilet or figlet in such ancient versions on the systems? What is the reason?

It’s the version currently available in Debian stable packages (https://packages.debian.org/stable/text/toilet), and installed as a dependency of a dependency of another package I use, neofetch.

**Comment by MickLesk** [comment]
Does neofetch running with the .tar.xz in our misc Project folder? Thats the latest figlet

**Comment by vsisl** [comment]
> I wonder why you have Toilet or figlet in such ancient versions on the systems? What is the reason?

I'm running Proxmox VE 8.3.2

I also have neofetch installed. So maybe it's indeed caused by neofetch? How could I update or install those fonts?

**Comment by onethree7** [comment]
problem:
```
root@jellyfin Dec 31 13:43 ~ > figlet -f slant "Tteck forever"
error: could not load font slant
```

wo:
`sudo wget -qO /usr/share/figlet/slant.flf http://www.figlet.org/fonts/slant.flf
`
works:
```
root@jellyfin Dec 31 13:42 ~ > figlet -f slant "Tteck forever"
  ________            __      ____                              
 /_  __/ /____  _____/ /__   / __/___  ________ _   _____  _____
  / / / __/ _ \/ ___/ //_/  / /_/ __ \/ ___/ _ \ | / / _ \/ ___/
 / / / /_/  __/ /__/ ,<    / __/ /_/ / /  /  __/ |/ /  __/ /    
/_/  \__/\___/\___/_/|_|  /_/  \____/_/   \___/|___/\___/_/     
```

the fonts are expected in "/usr/share/figlet/"

Can add a check in build.func (although i really would prefer NOT using any fonts , graphics, pictures and icons at all). I dont need fancy texts. All i need is a CLI and text in and out. The fancier, the more it breaks.

**Comment by MickLesk** [comment]
yes, ive time for 30min. I Prepare an Update, after merge, can you check it?

**Comment by onethree7** [comment]
im available.

**Comment by MickLesk** [comment]
can you check it?

**Comment by onethree7** [comment]
this fixed it for me.

- i was using the "update" method inside a LXC
- i had the dependency Toilet/Figlet version as shown above by others.
- my font was not included or placed correctly, 
- the updated build.func worked around this.

I think the assumption that "if run as INSTALL on the PVE" is fixed, can only be true as well.

**Comment by MickLesk** [comment]
I'll leave the issue open, maybe 1-2 others can still test it. But I think that every current use-case should be handled this way. 

So now it's PC rest, have a nice New Year's Eve ;)

**Comment by onethree7** [comment]
Happy new year , frohes neues :)

**Comment by vsisl** [comment]
For what it's worth, I fixed this bug by simply

```bash
apt install figlet
```

**Comment by MickLesk** [comment]
> For what it's worth, I fixed this bug by simply
> 
> ```bash
> apt install figlet
> ```

Its only possible with the correct repository
